projects/storefrontlib/src/cms-structure/outlet/provide-outlet.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  APP_INITIALIZER,
+  ComponentFactoryResolver,
+  FactoryProvider,
+  Type,
+} from '@angular/core';
+import { OutletPosition } from './outlet.model';
+import { OutletService } from './outlet.service';
+
+interface ProvideOutletOptions {
+  name: string;
+  position?: OutletPosition;
+  component: Type<any>;
+}
+
+export function provideOutlet(options: ProvideOutletOptions): FactoryProvider {
```

Comment:
Note for @ChristophHi: The util `provideOutlet` has been already merged to `develop` in a separate PR.